general/v0.1/registration.md

```diff
@@ -95,21 +89,16 @@ This validation MUST include:
 ![A bare-bones web form to add a provider with a textarea for the registration token](../../images/add_provider.svg)
 
 After successful submission of this data, the fediverse software MUST
-persist it and initiate an OAuth 2.0 "client credentials" flow to
-obtain an access token. This token can then be used to authenticate
-subsequent API calls.
+persist it.
 
-If authorization was successful the instance MUST create and persist an
-OAuth 2.0 application representing the FASP, including a client ID
-and a secret.
+It MUST then generate a client ID and a secret key for the FASP
+analogous to how the FASP did for the fediverse software.
 
-It MUST communicate its base URL, the generated client ID and secret to
-the FASP using the client credentials API endpoint.
+It MUST communicate its base URL, the generated client ID and secret key
+to the FASP using the client credentials API endpoint.
```

Comment:
Wouldn't public key be enough to verify the HTTP Message Signatures? If you send the private key over the wire you're back to relying on HTTPS

Reply:
I do not claim to be an expert on any of this (in fact I am most definitely not), but I did read a lot in the past days/weeks. And one advice that came up repeatedly was to not to use asymmetric crypto unless you absolutely have to and are really sure you know what you are doing.

Yes, in this (and all other proprosals, including the original OAuth one) we would send the respective (symmetric) secrets over the wire. With OAuth that would happen repeatedly, every time a new access token is requested. In these new proposals this happens only once for each side, so two times total.

And we are already relying on HTTPS and if we feel we cannot, then I think we have a whole new set of problems to solve :confused: